### ISSUE TITLE
Fix docker builds...

### ISSUE THREAD

**Issue by espg** [geostacks/GeoStacks#10]
@whyjz Apparently, `mplleaflet` is not the same thing as `ipyleaflet`; installing `ipyleaflet` is complicated by the fact that it is an extension to jupyterlab instead of just a vanilla python package... looking into [jupyterwith](https://github.com/tweag/jupyterWith) for possible docker workaround (solution may also be of interest to @fperez )

**Comment by espg** [comment]
@whyjz Ok, so I think I have a solution that is at least close to working... newest commit edea268cc0f6180ee5c39eaba2165450d318ea11 has an updated `default.nix` that builds an environment that will load ipyleaflet ; there's an example notebook (`doc/leaflet.ipynb`) to test that it's working as expected. I still need to re-write the spatial index function to use @jonathan-taylor 's contains, since the build references libs that are incompatible with geopandas/decartes. Modifying the functions shouldn't take too long, I might be able to finish it tonight.

There is a caveat with the build; it only works with jupyterlab, not plain jupyter notebooks... @fperez do you know if this is expected? I'm not familiar enough with ipyleaflet to know if the extension is only for jupyterlab. I'm also not sure how exactly this translates on the docker/binder side of things-- hopefully it's possible to force a jupyterlab instance to launch instead of just a vanilla notebook?

**Comment by whyjz** [comment]
A quick comment here: yes it is possible to launch a jupyter lab instance through binder by changing the binder link a bit. Here's an example from the EZTrack project -- the key is to put **urlpath=lab/tree/** here:
https://mybinder.org/v2/gh/whyjz/EZ-FeatureTrack/HEAD?urlpath=lab/tree/notebooks/arcticworkshop_demo.ipynb

**Comment by espg** [comment]
@whyjz yeah, I know that's how it's supposed to work... it just doesn't, or at least isn't in this case:

https://mybinder.org/v2/gh/geostacks/GeoStacks/HEAD?urlpath=lab/tree/doc/examples/leaflet.ipynb

The above will build fine:

```
...
Successfully tagged 3i2li627.gra7.container-registry.ovh.net/binder/ovhbhub-geostacks-2dgeostacks-71a6e9:3df6eaf52c6d36cc2bc0e24ec5c9dc28eeb14cef
Pushing image
Pushing image
...
Pushing image
Successfully pushed 3i2li627.gra7.container-registry.ovh.net/binder/ovhbhub-geostacks-2dgeostacks-71a6e9:3df6eaf52c6d36cc2bc0e24ec5c9dc28eeb14cefBuilt image, launching...
Launching server...
....
```
...but fail to launch with 404 error =/ 

![image](https://user-images.githubusercontent.com/3896161/118026902-5ffa3500-b32f-11eb-9b0b-1afa046037f3.png)

Using just the notebook link works fine:

https://mybinder.org/v2/gh/geostacks/GeoStacks/HEAD?urlpath=tree/doc/examples/leaflet.ipynb

My guess is it is an issue with multiple jupyterlab installs? I think repo2docker tries to build a jupyterlab instance on it's own, separate from the custom one that is built, which then probably is a bit of an issue... we may have to do a link to a custom docker container... although it would be nice to avoid that if at all possible

**Comment by fperez** [comment]
BTW, I think this one can be closed, right? In #13 and #14 we covered some of the issues we were having and now the build works OK, as far as I can tell (even if it's slower than we'd want it to due to all the extra manual updating we're doing).

**Comment by espg** [comment]
Agreed!
